### PR TITLE
Add status bar

### PR DESCRIPTION
**Pull request by s0nea**
Still todo:
- Replace all the placeholder icons with actual icons
  - [x] Add Rancher Desktop icon
  - [x] Add Kubernetes svg
  - [x] Add Container engine icon -> Depends on https://github.com/rancher/icons/pull/69 
- [x] Replace the dark green color with a lighter green in the network status component (the small dot behind the status)
- [x] Take some screenshots and add them here

Fixes: https://github.com/rancher-sandbox/rancher-desktop/issues/5122

Screenshot_2023-11-27_12-34-06
Screenshot_2023-11-27_12-33-53
Screenshot_2023-11-27_12-28-18
Screenshot_2023-11-27_12-26-33
Screenshot_2023-11-27_12-25-53
Screenshot_2023-11-27_12-25-37
Screenshot_2023-11-27_12-25-08

### CODE REVIEW

File: pkg/rancher-desktop/components/StatusBarItem.vue
Comment (rak-phillip) [inline pkg/rancher-desktop/components/StatusBarItem.vue:68]: I'm of the mind that slots should be exposed by default inside of a component, otherwise we end up in a scenario where some magic internal component logic can disable a slot unexpectedly.

**Review by rak-phillip — CHANGES_REQUESTED** [review]
Overall, I like the design of this and I think you did you great work. 

I am noticing that we're able to hit an odd breakpoint where the footer extends beyond the viewport and content gets cut off. This happens while the progress bar is present. 

image

File: pkg/rancher-desktop/components/StatusBarItem.vue
Comment (s0nea) [reply pkg/rancher-desktop/components/StatusBarItem.vue:40]: I tried it out. I think it's a good improvement. Thanks :slightly_smiling_face:

**Comment by s0nea** [comment]
> Overall, I like the design of this and I think you did you great work.
> 
> I am noticing that we're able to hit an odd breakpoint where the footer extends beyond the viewport and content gets cut off. This happens while the progress bar is present.

@rak-phillip thank you for your review! I updated the PR according to your comments.

File: pkg/rancher-desktop/assets/translations/en-us.yaml
Comment (rak-phillip) [inline pkg/rancher-desktop/assets/translations/en-us.yaml:143]: I'm fine with using "CE" in the status bar to save space, but we never specify what this abbreviation means anywhere in Rancher Desktop. Perhaps, we can specify "Container Engine: moby" or "Container Engine(CE): Moby" in the popover? This will provide people a quick way to see what CE stands for if it's not immediately obvious from the outset.

image

File: pkg/rancher-desktop/components/StatusBarItem.vue
Comment (rak-phillip) [inline pkg/rancher-desktop/components/StatusBarItem.vue:42]: The mouse cursor should be default instead of the I-beam for each status item. I think that we should also disable selection.

File: pkg/rancher-desktop/components/Version.vue
Comment (rak-phillip) [inline pkg/rancher-desktop/components/Version.vue:30]: We shouldn't require the `$data` instance property here and the shorthand `this.version` should suffice. 

```suggestion
      this.version = version;
```

File: pkg/rancher-desktop/components/NetworkStatus.vue
Comment (rak-phillip) [inline pkg/rancher-desktop/components/NetworkStatus.vue:65]: We might want to consider if we can just change the color of the icon to red/green instead of having the dot 🤔 

image

I don't think that we want to change anything for this PR, but we might want to raise the idea during a design discussion.

**Review by rak-phillip — CHANGES_REQUESTED** [review]
This is looking great ✨

I noticed a few more items that we can improve. Otherwise, I think we're nearly ready to merge.

File: pkg/rancher-desktop/components/NetworkStatus.vue
Comment (s0nea) [reply pkg/rancher-desktop/components/NetworkStatus.vue:65]: Oh... yes, indeed. @manuelecarlini and I talked already about it. There are two possible downsides:

1. If the icon is not shown, we don't have a colored indicator
2. It might work well on the dark theme. I'm not too sure about the light one. Maybe with the globe icon it's better than with the placeholder icon before.

The green color code is not final, yet. Once it is, I'll try it out and prepare some screenshots for discussion.

Comment (jandubois) [reply pkg/rancher-desktop/components/NetworkStatus.vue:65]: We could also make the word `online` green and `offline` red.

Whatever we do, colour should not be the only way we specify the value, so people with colour-vision impairment can still figure out the status without having to go to the hover tip.

File: pkg/rancher-desktop/assets/translations/en-us.yaml
Comment (jandubois) [reply pkg/rancher-desktop/assets/translations/en-us.yaml:143]: I think `Container Engine: moby` is fine; no need to repeat the `CE` abbreviation in the hover tip.

File: pkg/rancher-desktop/components/StatusBarItem.vue
Comment (s0nea) [reply pkg/rancher-desktop/components/StatusBarItem.vue:42]: I'll change the good accordingly. Although I'm not 100% sure about it. I can imagine it's neat to be able to copy e.g. the Rancher Desktop version for a bug report. Maybe we can discuss further options at some point. The change is trivial and easy to revert/adapt if we decide.

File: pkg/rancher-desktop/components/NetworkStatus.vue
Comment (s0nea) [reply pkg/rancher-desktop/components/NetworkStatus.vue:65]: I created an issue for further discussion here: https://github.com/rancher-sandbox/rancher-desktop/issues/6046

**Comment by s0nea** [comment]
> This is looking great ✨
> 
> I noticed a few more items that we can improve. Otherwise, I think we're nearly ready to merge.

Thank you for the suggestions! I updated the PR accordingly.

File: pkg/rancher-desktop/components/StatusBarItem.vue
Comment (rak-phillip) [reply pkg/rancher-desktop/components/StatusBarItem.vue:56]: I think you're right, we should leave the cursor as-is and allow for selection in the status bar. I think that supporting users so that they can quickly copy is enough of a reason to keep things as-is. 

We can follow-up to see if we want to implement some sort of "click to copy" mechanism, but that would definitely be an additional requirement if we choose to go that direction.

**Review by rak-phillip — DISMISSED** [review]
Looks great 🚀

I'll leave it up to you as to how we want to deal with the cursor/selection before we merge:

https://github.com/rancher-sandbox/rancher-desktop/pull/5985#discussion_r1404215490

**Comment by s0nea** [comment]
> Looks great 🚀
> 
> I'll leave it up to you as to how we want to deal with the cursor/selection before we merge:
> 
> [#5985 (comment)](https://github.com/rancher-sandbox/rancher-desktop/pull/5985#discussion_r1404215490)

Thank you! :slightly_smiling_face: Just pushed the revert to the original behavior.

**Review by rak-phillip — APPROVED** [review]

**Pull request merged**
